unity-renderer/Assets/Scripts/Example.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System;
+using UnityEngine;
+
+namespace DefaultNamespace
+{
+    public enum Side { Left, Right }
+
+    public class Example : MonoBehaviour, IInterface
+    {
+
+        public const string ASSET_PATH = "AvatarUI";
+
+        private const int MAX_TIMER = 10;
+        public static readonly int WaveAnimHash = Animator.StringToHash("Wave");
+        private static readonly int danceAnimHash = Animator.StringToHash("Dance");
+
+        public int SomeField01;
+
+        [HideInInspector] public int SomeField02;
+
+        [SerializeField] private Collider collider;
+        [NonSerialized] public int SomeField03;
+
+        internal bool someField1; // Pascal or Camel ?
+        private int someField14;
+        protected int someField2;
+
+        public bool Property1 { get; set; }
+        public bool Property2 { private get; set; }
+        public bool Property3 { get; }
+        public bool Property4 { get; private set; }
+
+        private void Awake()
+        {
+
+            // GetComponent
+        }
+
+        private void OnEnable()
+        {
+            // subscribe?
+        }
+
+        private void OnDisable()
+        {
+            // unsubscribe?
+        }
+
+        private void OnCollisionEnter(Collision collision) { }
```

Comment:
Fully agree with step by step transition. That is why I tried to not change the naming convention at all but just document it. Because re-naming could be quite a huge shift (as you mention about PRs). I was thinking that fixing the ordering could be the first step (which will improve navigation a lot).

However, answering your question, I would rather prefer to have `_` for all non-public Fields and leave the local Variables and methods Parameters with the `camelCase`. It is more of a standard in the industry. I think @Antivortex also likes `_`-for-fields solution. I see `_` as an additional tool, but in a small scope of the method's body you do not need such tools - everything is on the screen.

Reply:
Sure! I dont really mind as long as we differentiate them.

Reply:
i would also avoid `_`.. i prefer to referrence fields as `this.field` but rather rename the parameter so it does not collide

Reply:
when defining code style for an open-source project, it could be beneficial to think about what is most commonly used notation, so it will be smoother onboarding for the external contributors. That is why I would vote for [Google C#](https://google.github.io/styleguide/csharp-style.html) and [.NET code style](https://learn.microsoft.com/en-us/dotnet/csharp/fundamentals/coding-style/coding-conventions), where _ is used for `private`/`protected` fields. 

However, if create our own style and agree on `_` for parameters, then @Antivortex question is valid - should we use this notation also for local variables (variables defined inside the method)?

Reply:
Unless you wanna go the Hungarian Notation way (which I truly dislike) there's stuff we have to give up.

You have public <> private <> params <> local variables and not so many permutation of styles. Nowadays IDEs do the heavy lifting and with a good coloring schema that's usually just enough to know what you are dealing with.
I think it's just a matter of prioritizing what we want to stand out.

To be honest @popuz has got a really good point a bout using the most commonly used style. .NET code style seems like the correct way IMO.